astropy/table/groups.py

```diff
@@ -256,17 +256,18 @@ def aggregate(self, func):
                     vals = func.reduceat(par_col, i0s)
             else:
                 vals = np.array([func(par_col[i0: i1]) for i0, i1 in zip(i0s, i1s)])
+            out = par_col.__class__(vals)
         except Exception as err:
-            raise TypeError("Cannot aggregate column '{}' with type '{}'"
-                            .format(par_col.info.name,
-                                    par_col.info.dtype)) from err
-
-        out = par_col.__class__(data=vals,
-                                name=par_col.info.name,
-                                description=par_col.info.description,
-                                unit=par_col.info.unit,
-                                format=par_col.info.format,
-                                meta=par_col.info.meta)
+            raise TypeError("Cannot aggregate column '{}' with type '{}': {}"
+                            .format(par_col.info.name, par_col.info.dtype, err)) from err
+
+        out_info = out.info
```

Comment:
Interesting point. Doing that should work, but will be slower by some unknown amount because it goes through more generalized logic to figure out which attributes to copy. Maybe it's better to be slower but correct and future proof?

Reply:
I think if it passes the tests with it, I would tend to just use it. That way, any class that defines additional attributes will just work (and we do the same for slicing, which is reasonably similar to this case).

Reply:
Actually this does not work. Having a look at the logic in `__set__` in `DataInfo`, the copying of attributes is strictly limited to those which live in the `_attrs` dict. It specifically avoids attributes that live on the parent, which in the case of `Column` is all of them. This behavior is a little surprising to me even if I probably wrote that code.

Maybe we can revisit this but it is always scary to change something so deep in the code. In any case the code for this PR is fine.